ister_gui.py

```diff
@@ -1086,45 +1121,19 @@ def handler(self, config):
             # re-build the widgets
             return self._action
 
-        self.error = ''
-
-        if self.static_check:
-            try:
-                ipaddress.ip_address(self.gateway_e.get_edit_text())
-            except Exception as err:
-                if self._action not in ['Next', 'Previous']:
-                    self.error = 'The configuration for "gateway" is invalid '\
-                                 '{}'.format(err)
-                    # an empty action signals to refresh the page
-                    self._action = ''
-                    self.static_check = False
-
-            try:
-                ipaddress.ip_address(self.static_ip_e.get_edit_text())
-            except Exception as err:
-                if self._action not in ['Next', 'Previous']:
-                    self.error = 'The configuration for "ip" is invalid {}'\
-                                 .format(err)
-                    # an empty action signals to refresh the page
-                    self._action = ''
-                    self.static_check = False
-
-            # this will overwrite the ip error if it was created
-            if self.interface_e.get_edit_text() not in self.ifaceaddrs.keys() \
-                    and self._action not in ['Next', 'Previous']:
-                self.error = 'Interface "{}" not detected'\
-                             .format(self.interface_e.get_edit_text())
-                # an empty action signals to refresh the page
-                self._action = ''
-                self.static_check = False
-
-        if not self.error:
-            config = self.config
-            return self._action
+        if self.static_ready:
+            # go back two original widgets (through both Padding and AttrMap)
+            # to set the label of the button.
+            self.static_button.original_widget.original_widget.set_label(
+                    ('button', 'Set static IP configuration'))
         else:
-            Alert('Error!', self.error).do_alert()
-            self.error = ''
-            return self._action
+            self.static_button.original_widget.original_widget.set_label(
+                    ('ex', 'Set static IP configuration'))
+
+        # proxy settings may have been added during the self.run_ui() step,
+        # copy our config instance back to the config.
+        config = self.config
+        return self._action
```

Comment:
So I'm going to merge this as is. We can go back later and change things if we get feedback.